pkg/pillar/zedcloud/wstunnelclient.go

```diff
@@ -394,26 +392,17 @@ func (wsc *WSConnection) processResponses() {
 	log.Infof("Processing responses from local relay: %s", host)
 
 	var id int64
-	for {
-		select {
-		case <-wsc.tun.requestSentChan:
-
-			if err := wsc.refreshLocalConnection(host, false); err != nil {
-				log.Errorf("Error encountered while refreshing local connection: %s", err.Error())
-				break
-			}
-			wsc.localConnection.SetReadDeadline(time.Now().Add(500 * time.Millisecond))
-			responseBuffer := make([]byte, 524288)
-			responseBuffer, _ = ioutil.ReadAll(wsc.localConnection)
-			num := len(responseBuffer)
-			if num > 0 {
-				response := responseBuffer[:num]
-				log.Debugf("[id=%d] Read local connection payload: \"%s\"", id, string(response))
-
-				wsc.writeResponseMessage(id, bytes.NewBuffer(response))
-				id++
-			}
-		default:
+	ticker := time.NewTicker(100 * time.Millisecond)
```

Comment:
Why poll every 100 ms? That is slow too.

Doesn't the wsc.localConnection.Read() block until some data is available?

Reply:
> Why poll every 100 ms? That is slow too.
> 
I tested lesser (50ms) and more (200ms) intervals, there was no visible boost by reducing the polling interval further. So the idea was to hit a sweet spot without being too busy on resources.
> Doesn't the wsc.localConnection.Read() block until some data is available?
>
Yes, this was the intended implementation during the change. But during testing it was observed that `guacd` returns an empty response to the relay everytime you send a read.